packages/mask/src/utils/i18n-next-ui.ts

```diff
@@ -3,11 +3,17 @@ import type en from '../../shared-ui/locales/en-US.json'
 import type { i18NextInstance, TranslateOptions } from '@masknet/shared-base'
 import { SupportedLanguages } from '@masknet/public-api'
 
-export type I18NFunction = <TKeys extends keyof typeof en>(
+type PluralsSuffix = 'zero' | 'one' | 'two' | 'few' | 'many' | 'other'
+
+type LocaleKeys = keyof typeof en
+type ExtendBaseKeys<K> = K extends `${infer B}$${string}` ? B | K : K extends `${infer B}_${PluralsSuffix}` ? B | K : K
+type AvailableLocaleKeys = ExtendBaseKeys<LocaleKeys>
+
+export type I18NFunction = <TKeys extends AvailableLocaleKeys>(
```

Comment:
I suggest to stop using this, use i18n exported from `packages/mask/shared-ui/locales`, it provides context support.